pkg/cmd/backuprestore/backupserver.go

```diff
@@ -2,21 +2,45 @@ package backuprestore
 
 import (
 	"context"
-	"github.com/spf13/pflag"
+	"fmt"
 	"os"
 	"os/signal"
 	"syscall"
+	"time"
 
-	"github.com/spf13/cobra"
 	"k8s.io/klog/v2"
+
+	cron "github.com/robfig/cron/v3"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 )
 
+const backupVolume = "/var/lib/etcd-auto-backup"
+
 var shutdownSignals = []os.Signal{os.Interrupt, syscall.SIGTERM}
 
+type backUpEr interface {
+	runBackup(opts *backupOptions) error
+}
+
+type backUpErImpl struct{}
```

Comment:
sounds Java-esque, but I also don't have a better name

```suggestion
type backRunnerImpl struct{}
```